string id
int32 fixedPoseIndex
int32 constrainedPoseIndex